src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/rules/MediationSuccessfulRule.java

```diff
@@ -0,0 +1,63 @@
+package uk.gov.hmcts.cmc.claimstore.services.ccd.callbacks.rules;
+
+import org.apache.commons.io.FilenameUtils;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.cmc.ccd.domain.CCDCase;
+import uk.gov.hmcts.cmc.ccd.domain.CCDClaimDocument;
+import uk.gov.hmcts.cmc.ccd.domain.CCDClaimDocumentType;
+import uk.gov.hmcts.cmc.ccd.domain.CCDCollectionElement;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+@Component
+public class MediationSuccessfulRule {
+    public static final String STAFF_UPLOAD_MEDIATION_AGREEMENT =
+        "Upload Mediation Agreement";
+    public static final String STAFF_UPLOAD_TYPE_MEDIATION_AGREEMENT =
+            "You need to upload a document of type Mediation agreement";
+    public static final String STAFF_UPLOAD_PDF_MEDIATION_AGREEMENT =
+            "Mediation agreement needs to be of type PDF";
+
+    public List<String> validateMediationAgreementUploadedByCaseworker(CCDCase ccdCase) {
```

Comment:
This method could use revisiting. You can get different results based on the order the documents are in.